template/src/store/reducers/app.ts

```diff
@@ -14,24 +13,26 @@ const appSlice = createSlice({
   name: 'app',
   initialState,
   reducers: {
-    [APP_CONSTANTS_ACTIONS.GET_SETTING_APP_ACTIONS.HANDLER]: (state: IApp, action: IAction): void => {},
-    [APP_CONSTANTS_ACTIONS.GET_SETTING_APP_ACTIONS.SUCCESS]: (state: IApp, action: IAction): void => {},
-    [APP_CONSTANTS_ACTIONS.GET_SETTING_APP_ACTIONS.FAILURE]: (state: IApp, action: IAction): void => {},
-    [APP_CONSTANTS_ACTIONS.LOGIN_ACTIONS.HANDLER]: (state: IApp, action: IAction): void => {},
-    [APP_CONSTANTS_ACTIONS.LOGIN_ACTIONS.SUCCESS]: (state: IApp, action: IAction): void => {},
-    [APP_CONSTANTS_ACTIONS.LOGIN_ACTIONS.FAILURE]: (state: IApp, action: IAction): void => {},
-    getSettings: (state: IApp, action: IAction): void => {},
-    setAppStack: (state: IApp, action: IAction): void => {
+    // TODO: custom any in PayloadAction<any> depend on action type
+    [APP_CONSTANTS_ACTIONS.GET_SETTING_APP_ACTIONS.HANDLER]: (state): string => {
+      return state.appState;
+    },
+    [APP_CONSTANTS_ACTIONS.GET_SETTING_APP_ACTIONS.SUCCESS]: (state, action: PayloadAction<any>): void => { },
+    [APP_CONSTANTS_ACTIONS.GET_SETTING_APP_ACTIONS.FAILURE]: (state, action: PayloadAction<any>): void => { },
+    [APP_CONSTANTS_ACTIONS.LOGIN_ACTIONS.HANDLER]: (state, action: PayloadAction<any>): void => { },
+    [APP_CONSTANTS_ACTIONS.LOGIN_ACTIONS.SUCCESS]: (state, action: PayloadAction<any>): void => { },
+    [APP_CONSTANTS_ACTIONS.LOGIN_ACTIONS.FAILURE]: (state, action: PayloadAction<any>): void => { },
+    getSettings: (): void => { },
+    setAppStack: (state, action: PayloadAction<string>): void => {
       state: state.appState = action.payload
     },
-    getSettingsSuccess: (state: IApp, action: IAction): void => {},
-    setShowGlobalIndicator: (state: IApp, action: IAction): void => {
+    setShowGlobalIndicator: (state, action: PayloadAction<boolean>): void => {
       state: state.showGlobalIndicator = action.payload
```

Comment:
Change state: state.showGlobalIndicator = action.payload to state.showGlobalIndicator

Reply:
Updated